### PR TITLE
Correctly assign kill data to mine defuser

### LINKED ISSUE #15: Defusal Without Death
With respect to the ePlayerDieEvent [here, before pull request](https://github.com/allejo/UselessMine/blob/master/UselessMine.cpp#L269).

I think that there might be an issue here. This case checks the mines when a player dies; this includes defused mines. In the typical case, this is fine: a player defuses a mine, and the mine goes off at the owner's location, the owner dies, and everything's nice. The problem would be if the defused mine SW goes off but *doesn't* kill the owner (for instance, if the owner is moving quickly or holding SH). When the owner *does* die, this message will trigger at that point, giving a seemingly random message. Note also that this means that the player won't get a message when the mine is defused, but when the owner dies. 

A quick logic walkthrough:

Owner lays a mine, and hops on the tracks around the map.
Victim with BD runs over the mine, which triggers on the track.
The SW misses the owner, who proceeds to drop off the track and get shot. 
Note that the mine was flagged as "defused", so when the ePlayerDieEvent triggers, the defused section will activate. The killer ID is set to the mine defuser, and the messages are sent.

So in summary, if the owner evades the defusal the messages will still be sent. I don't know if the points will be misassigned.

### PR DESCRIPTION
**Pull request by asinck**
This fixes #15 

This also sends a message when the owner is killed by their own mine.

### CODE REVIEW

**Comment by asinck** [comment]
*should fix issue 15 :)

File: UselessMine.cpp [hunk 1]
```diff
@@ -340,13 +340,21 @@ void UselessMine::Event (bz_EventData *eventData)
                             bz_sendTextMessagef(BZ_SERVER, playerID, "You were killed by %s's mine", owner);
                         }
                     }
+                    else
+                    {
+                        //if the owner was killed with their own mine, send a message
```
Comment (allejo) [inline UselessMine.cpp]: Space after the `//`

File: UselessMine.cpp [hunk 2]
```diff
@@ -340,13 +340,21 @@ void UselessMine::Event (bz_EventData *eventData)
...
+                        bz_sendTextMessagef(BZ_SERVER, BZ_ALLUSERS, "%s was owned by their own mine!", owner);
+                    }
                 }
                 else if (mine.defused)
                 {
-                    dieData->killerID = mine.defuserID;
-
-                    if (playerID != mine.owner)
+                    // Make sure the killer was the server, and the victim was the owner.
+                    if (dieData->killerID == 253 && playerID == mine.owner)
```
Comment (allejo) [inline UselessMine.cpp]: Wouldn't checking `playerID == mine.owner` ignore any bystanders who happened to be near the defusal SW?

File: UselessMine.cpp (hunk 1 above)
Comment (asinck) [reply UselessMine.cpp]: Whoops, missed that one.

File: UselessMine.cpp (hunk 2 above)
Comment (asinck) [reply UselessMine.cpp]: oops

**Comment by asinck** [comment]
I made a new (hard-coded) message for bystander victims of a defusal.

File: UselessMine.cpp [hunk 3]
```diff
@@ -340,30 +340,49 @@ void UselessMine::Event (bz_EventData *eventData)
...
+                    if (dieData->killerID == 253)
                     {
-                        if (!defusalMessages.empty())
+                        // Since the killer was the server, it was the defusal that killed the player. Thus,
+                        // give credit where credit is due.
+                        dieData->killerID = mine.defuserID;
+
+                        // Take note of the defuser's callsign
+                        const char* defuser = bz_getPlayerCallsign(mine.defuserID);
+
```
Comment (allejo) [inline UselessMine.cpp]: Trim the whitespace on this line

File: UselessMine.cpp [hunk 4]
```diff
@@ -340,30 +340,49 @@ void UselessMine::Event (bz_EventData *eventData)
...
+                        
+                        if (playerID == mine.owner)
                         {
-                            // The random number used to fetch a random taunting defusal message
-                            int randomNumber = rand() % defusalMessages.size();
-
-                            // Get a random defusal message
-                            std::string defusalMessage = defusalMessages.at(randomNumber);
-                            bz_sendTextMessage(BZ_SERVER, BZ_ALLUSERS, formatMineMessage(defusalMessage, owner, bz_getPlayerCallsign(mine.defuserID)).c_str());
+
```
Comment (allejo) [inline UselessMine.cpp]: whitespace here too

**Comment by allejo** [comment]
Hmm, so on two separate bomb defusals, these are the messages I got. Kinda iffy on the consistency.

```
SERVER: the map successfully defused allejo's mine!
allejo: blew myself up
SERVER: allejo was owned by their own mine!
```

and

```
SERVER: the map successfully defused allejo's mine!
allejo: felt the effects of the map's shockwave
SERVER: allejo successfully defused the map's mine!
```

File: UselessMine.cpp (hunk 3 above)
Comment (asinck) [reply UselessMine.cpp]: Fixed, unpushed.

File: UselessMine.cpp (hunk 4 above)
Comment (asinck) [reply UselessMine.cpp]: Fixed, unpushed.

**Pull request merged**